### PR TITLE
Improve test wrapper

### PR DESCRIPTION
**Pull request by JoelSpeed**
Last night I realised there were a couple of problems with the test wrapper, this PR fixes them

Firstly, we would not parse the output of a failed test run because we were exiting too early. I have moved the error check for this so that we still parse the output even if the `go test` fails.

Secondly, if you call `go test` multiple times in a job, it would fail. We need to create separate files per call to go test, so I've made a function to check for the existence of files and give the next `junit_<x>.xml` file available (with a test for this)

### CODE REVIEW

File: images/golang-builder/wrapper/main.go [hunk 1]
```diff
@@ -161,3 +165,18 @@ func contains(s []string, e string) bool {
 	}
 	return false
 }
+
+// getJUnitFileName returns the next available file name for files in the given
+// directory with the naming pattern of junit_<x>.xml where <x> is an increasing
+// integer
+func getJUnitFileName(dir string) string {
+	i := 0
```
Comment (gargath) [inline images/golang-builder/wrapper/main.go]: This is going to be a signed int and as such could overflow and wrap around if no free filenames are available...

I feel we should either use something like a timestamp that can't clash or guard against the case where we've run out of numbers.

**Review by wonderhoss — CHANGES_REQUESTED** [review]

File: images/golang-builder/wrapper/main.go (hunk 1 above)
Comment (JoelSpeed) [reply images/golang-builder/wrapper/main.go]: > This is going to be a signed int and as such could overflow and wrap around if no free filenames are available...

You would have to run `go test` a lot of times within a single CI pipeline for that to happen though, would you be more comfortable if I explicitly made it an `int64`? Alternatively, as you suggest, I can look into timestamping it, but that doesn't guarantee uniqueness strictly

**Review by wonderhoss — APPROVED** [review]
👍

**Pull request merged**